test/integration/azureSpecialProperties.spec.ts

```diff
@@ -33,16 +33,26 @@ describe("auth validation", () => {
     const client = new AzureSpecialPropertiesClient(
       mockCredential,
       "1234-5678-9012-3456",
-      { allowInsecureConnection: true }
+      {
+        allowInsecureConnection: true,
+        credential: mockCredential,
```

Comment:
Why do we have to pass the credential as part of the ClientOptions? We are already passing it as the constructor's first parameter

Reply:
Modified the code now. Also created https://github.com/Azure/autorest.typescript/issues/952 to fix the user agent. (Because that needs changes to the core)